SLEEP-1337 Add support for mobile bulk uploads

bramj opened the pull request:
In Trypelim, we have performance issues on the form uploads.
To improve the performance of the uploads, we're grouping forms and org units into zip files (batches of 2000 forms) and upload these with only 1 API call per zip file.

This PR adds a job to process these bulk uploads:
1. A new API endpoint `/api/mobile/bulkupload/` that takes a `zip_file`. This endpoint simply uploads the file to S3, launches an async job, and returns HTTP 204.
2. The process job creates a record in the `APIImport` for future reference (and to be able to tag as `has_problem`) and processes all files in the zip in a db transaction.

Related JIRA tickets : https://bluesquare.atlassian.net/browse/SLEEP-1337

## Self proofreading checklist

- ~~[ ] Did I use eslint and black formatters~~
- [x] Is my code clear enough and well documented
- ~~[ ] Are my typescript files well typed~~
- [ ] New translations have been added or updated if new strings have been introduced in the frontend
- [x] My migrations file are included
- [x] Are there enough tests
- [ ] Documentation has been included (for new feature)

![image](https://github.com/BLSQ/iaso/assets/2719739/44ff62e7-5d29-44f9-bfdf-90f5951615a7)

bmonjoie committed:
SLEEP-1337 Spike to receive a zip for synchronization

bramj committed:
SLEEP-1337 Bulk upload: Move zip processing to worker

Upload file to S3 and pass the object_name to the worker process.
The worker will download the file and process it accordingly.

bramj committed:
SLEEP-1337 Bulk upload v1

bramj committed:
SLEEP-1337 Force use of TemporaryFileUploadHandler

See https://docs.djangoproject.com/en/4.2/topics/http/file-uploads/#upload-handlers

> By default, if an uploaded file is smaller than 2.5 megabytes, Django will hold the entire contents of the upload in memory. This means that saving the file involves only a read from memory and a write to disk and thus is very fast.

For our upload to S3, we want to force to always use the
TemporaryFileUploadHandler. If not, the method temporary_file_path()
that we use does not exist.

bramj committed:
SLEEP-1337 Add APIImport record + db transaction...

for the process_mobile_bulk_upload task.
Also add a first test for the happy path.

bramj committed:
SLEEP-1337 Support .webp image format in frontend

bramj committed:
SLEEP-1337 Improve bulk process job and it's test

bramj committed:
SLEEP-1337 Refactoring, improve Task message

bramj committed:
SLEEP-1337 Improve task finish message

bramj commented on iaso/tasks/process_mobile_bulk_upload.py:
```diff
@@ -0,0 +1,163 @@
+    return instance_files
+
+
+# Trypelim-specific
```
Main issue I see for merging this is how this code is Trypelim specific (although it won't affect any form that doesn't have a `serie_id` attribute)

bramj commented:
@mathvdh assigning you for review, I think Martin is too busy at the moment

madewulf commented on iaso/api/mobile/bulk_uploads.py:
```diff
@@ -0,0 +1,95 @@
+    def validateZipFile(self):
+        if self.is_valid():
+            return self.validated_data["zip_file"]
+        raise ValueError("Zip file not valid")
```
Does this do something for real, outside of checking that there is a file?

bramj replied:
No, just a check of the file

madewulf commented on iaso/tasks/process_mobile_bulk_upload.py:
```diff
@@ -0,0 +1,163 @@
+    user = User.objects.get(id=user_id)
+    project = Project.objects.get(id=project_id)
+
+    api_import = APIImport.objects.create(
```
I would have said that this should be the very first thing happening with the zip file, before any check on the user or other treatments. The goal is to store the data before any other bug can happen, and there, it seems to happen relatively late (after even the pickup by the worker if I understand correctly, which means that if the worker is off, the payload can be lost?)

bramj replied:
The file is uploaded to S3 before being passed to the worker, so we wouldn't really lose the data. That said, you're totally right we should create the import object first, which will make it a lot easier to track when something goes wrong. I'll fix it.

madewulf commented on iaso/tasks/process_mobile_bulk_upload.py:
```diff
@@ -0,0 +1,163 @@
+    )
+
+
+def process_instance_xml(directory, zip_ref, user):
```
Isn't this duplicated code from /sync/views.py 

In this case, we should make it a function as to ensure that when we update the code somewhere with new features, we don't forget to update it everywhere

bramj replied:
:+1: agreed, I've extracted a function

bramj committed:
SLEEP-1337 Improvement: Create APIImport record...

as a first step in the view.
This way we keep an easier trace of failed calls.

Note: I did not use the @safe_api_import decorator, because I want to
pass the api_import instance along to the worker. Since in the decorator
it is created "around" the view method, and since it is wrapped in a db
transaction, this can give weird behaviour with the record not yet
existing in the db.

bramj committed:
SLEEP-1337 Refac: Extract function for XML instance creation

bramj committed:
SLEEP-1337 Fix: Exception handling on bulk upload endpoint

bramj committed:
SLEEP-1337 Add several tests for bulk upload API endpoint
